src/Microsoft.VisualStudio.Threading/WeakKeyDictionary.cs

```diff
@@ -67,6 +67,17 @@ public int Count
             get { return this.dictionary.Count; }
         }
 
+        /// <summary>
+        /// Gets all key values in the dictionary.
+        /// </summary>
+        internal IEnumerable<TKey> Keys
```

Comment:
>IEnumerable [](start = 17, length = 11)

Why type this with the interface, when `KeyEnumerable` itself is internal? Should the nested type be private, or should we expose this as a concrete type?
And if as a concrete type, should we keep it as a struct to avoid the allocation?

Reply:
Yes, it was left behind in a previous change.  I will change it to private.